### PR TITLE
improve: exim: Address SAST (sonar) findings

### PR DESCRIPTION
**Pull request by kingthorin**
- Remove static modifier on enum, which is implicitly static.
- Return logical comparisons directly instead of within another conditional.
- Add private constructor to utility class (to prevent java from implicitly offering a public constructor).

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarImporter.java [hunk 1]
```diff
@@ -49,6 +49,10 @@
 
     private static ExtensionHistory extHistory;
 
+    private HarImporter() {
+        throw new IllegalStateException("Utility class");
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarImporter.java:53]: Not necessary.

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/har/PopupMenuItemSaveHarMessage.java [hunk 2]
```diff
@@ -122,13 +122,9 @@ public SaveHarFileChooser() {
                     new FileFilter() {
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/har/PopupMenuItemSaveHarMessage.java:122]: Worth extracting a class (duplicated in both menus).

**Comment by kingthorin** [comment]
Tweaked

**Comment by kingthorin** [comment]
Done (I think)

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.io.File;
+import javax.swing.filechooser.FileFilter;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+public class EximFileChooser extends WritableFileChooser {
+
+    private static final long serialVersionUID = 495386048962640141L;
+    String fileExtension;
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java:30]: `private final`

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                fileName += fileExtension;
+                setSelectedFile(new File(fileName));
+            }
+        }
+
+        super.approveSelection();
+    }
+}
+
+class EximFileFilter extends FileFilter {
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java:53]: `private static` (and move inside the above class).

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    String fileExtension;
+    String fileDescription;
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java:56]: `private final`

File: addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    public EximFileFilter(String fileExtension, String fileDescription) {
```
Comment (thc202) [inline addOns/exim/src/main/java/org/zaproxy/addon/exim/EximFileChooser.java:58]: Remove public modifier.

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**